packages/protocol/contracts/stability/GrandaMento.sol

```diff
@@ -152,32 +197,167 @@ contract GrandaMento is
+      "Sender cannot cancel the exchange proposal"
+    );
+    // Mark the proposal as cancelled. Do so prior to refunding as a measure against reentrancy.
+    proposal.state = ExchangeProposalState.Cancelled;
+    // Remove the proposal from the active list.
+    activeProposalIds.remove(proposalId);
+    // Get the token and amount that will be refunded to the proposer.
+    (IERC20 refundToken, uint256 refundAmount) = getSellTokenAndSellAmount(proposal);
+    // Finally, transfer out the deposited funds.
+    require(
+      refundToken.transfer(proposal.exchanger, refundAmount),
+      "Transfer out of refund token failed"
+    );
+    emit ExchangeProposalCancelled(proposalId, msg.sender);
+  }
+
+  /**
+   * @notice Executes an exchange proposal that's been approved and not vetoed.
+   * @dev Callable by anyone. Reverts if the proposal is not in the Approved state
+   * or vetoPeriodSeconds has not elapsed since approval.
+   * @param proposalId The identifier of the proposal to execute.
+   */
+  function executeExchangeProposal(uint256 proposalId) external nonReentrant {
+    ExchangeProposal storage proposal = exchangeProposals[proposalId];
+    // Require that the proposal is in the Approved state.
+    require(proposal.state == ExchangeProposalState.Approved, "Proposal must be in Approved state");
+    // Require that the veto period has elapsed since the approval time.
+    require(
+      proposal.approvalTimestamp.add(vetoPeriodSeconds) <= block.timestamp,
+      "Veto period not elapsed"
+    );
+    // Mark the proposal as executed. Do so prior to exchanging as a measure against reentrancy.
+    proposal.state = ExchangeProposalState.Executed;
+    // Remove the proposal from the active list.
+    activeProposalIds.remove(proposalId);
+
+    // Perform the exchange.
+    (IERC20 sellToken, uint256 sellAmount) = getSellTokenAndSellAmount(proposal);
+    // If the exchange sells CELO, the CELO is sent to the Reserve from this contract
+    // and stable token is minted to the exchanger.
+    if (proposal.sellCelo) {
+      // Send the CELO from this contract to the reserve.
+      require(
+        sellToken.transfer(address(getReserve()), sellAmount),
+        "Transfer out of CELO to Reserve failed"
+      );
+      // Mint stable token to the exchanger.
+      require(
+        IStableToken(proposal.stableToken).mint(proposal.exchanger, proposal.buyAmount),
+        "Stable token mint failed"
+      );
+    } else {
+      // If the exchange is selling stable token, the stable token is burned from
+      // this contract and CELO is transferred from the Reserve to the exchanger.
+
+      // Burn the stable token from this contract.
+      require(IStableToken(proposal.stableToken).burn(sellAmount), "Stable token burn failed");
+      // Transfer the CELO from the Reserve to the exchanger.
+      require(
+        getReserve().transferExchangeGold(proposal.exchanger, proposal.buyAmount),
```

Comment:
The idea is to have Granda Mento in CR5, leaving it unusable as all the parameters are zero, and then make a follow up proposal for activation setting the limits, setting the approver and a `Reserve.addExchangeSpender` call?

If so for me it was useful to keep a checklist for the proposals and even write a draft json (although we still don't have a clear place to check those on the repo, there are some jsons in the celo-proposals repo)